presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java

```diff
@@ -80,7 +80,8 @@ private boolean isSupportedType(Type type)
 
         if (type instanceof ArrayType) {
             checkArgument(type.getTypeParameters().size() == 1, "expecting exactly one type parameter for array");
-            return isSupportedPrimitive(type.getTypeParameters().get(0));
+            return true;
```

Comment:
do you support array of maps and other complex nested types in arrays?

Reply:
Yes, I will be adding more unit tests.